### ISSUE TITLE
Electron/NodeJS App: Swap out IndexedDB with pure disk storage

### ISSUE THREAD

**Issue by ronak2121** [google/shaka-player#3614]
**Have you read the [Tutorials](http://shaka-player-demo.appspot.com/docs/api/tutorial-welcome.html)?**
Yes

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes; this is a related thread: https://github.com/google/shaka-player/issues/1005

**What version of Shaka Player are you using?**
3.2.0

**Please ask your question**
I have a usecase (in an Electron based app) to download assets and then be able to read the downloaded assets out of band from the app. I saw that you have an [interface](https://github.com/google/shaka-player/blob/master/lib/offline/storage.js) to abstract away the details of the storage mechanism.

Can we extend this to directly write to the file system, instead of a database? We'd like to be able to save the entire asset as a single file on disk, with all segments appended together (These are VOD audio only assets).

**Comment by ronak2121** [comment]
Hi, is there any update on this?

**Comment by ronak2121** [comment]
Pinging for a comment?

**Comment by joeyparrish** [comment]
Hi @ronak2121,

There is an abstraction interface, but we never made it "official" or built non-IDB-based plugins for that.  You can certainly build your own storage mechanism on top of that interface, though, in a fork, and we would be happy to advise as you make progress.

I'm not sure we want to own a special implementation for Electron, but if you're willing to maintain it, and it doesn't break our existing architecture, then we might be willing to merge a PR.

Does this help?

**Comment by ronak2121** [comment]
Yes. Is the abstraction interface the same one that I quoted above? Do we largely just have to implement the interface using our disk storage solution? If so, then it looks like a clean place for us to own our extension, without making any changes to your codebase, reducing the amount of maintenance on our end.

We'll see if we see a need to publicly submit a PR vs keeping this internal to us at a later date.
